library/core/src/result.rs

```diff
@@ -796,7 +818,8 @@ impl<T, E> Result<T, E> {
     /// ```
     #[inline]
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn map<U, F: FnOnce(T) -> U>(self, op: F) -> Result<U, E> {
+    #[rustc_const_unstable(feature = "const_result_callback_methods", issue = "67792")]
+    pub const fn map<U, F: ~const FnOnce(T) -> U + ~const Destruct>(self, op: F) -> Result<U, E> {
```

Comment:
When inline bounds get messy, it's cleaner to move them to `where`

```suggestion
    pub const fn map<U, F>(self, op: F) -> Result<U, E> {
    where
         F: ~const FnOnce(T) -> U + ~const Destruct
```